performances/performance_test/src/ros2/tracker.cpp

```diff
@@ -60,10 +61,12 @@ void performance_test::Tracker::scan(
                 ev.description = description.str();
                 elog->write_event(ev);
             }
+        } else {
+            std::cerr << "warning: order of message is not sequential! Is node_name unique for each publisher?" << std::endl;
         }
 
         // Check if the message latency qualifies the message as a lost or late message.
-        const int  period_us = 1000000 / _frequency;
+        const int  period_us = 1000000 / _frequency.mean();
```

Comment:
I'm not sure this is the correct approach. 

Assume there is a topic with 2 publishers and a dynamic size message type. 
One publisher is publishing small messages at a fast rate, the other publisher is publishing big messages at a much slower rate. 
Using the mean frequency may result in showing all the messages published by the second publisher as late, simply because they are not as fast as the ones published by the first.

Reply:
true, so instead of using a single frequency we should use a list based on the node_name again.
for the output to the log it is then not the mean but the sum of those values?